torch/utils/_sympy/functions.py

```diff
@@ -6,6 +6,7 @@
 
 import sympy
 from sympy import S
+from sympy.core.numbers import equal_valued
```

Comment:
This line is causing issue on my side:
```
Python 3.10.14 (main, May  6 2024, 19:42:50) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sympy
>>> from sympy.core.numbers import equal_valued
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'equal_valued' from 'sympy.core.numbers'
```
My python version is 3.10 and sympy version is 1.13.1